libvast/vast/format/json.hpp

```diff
@@ -58,6 +58,8 @@ struct default_selector {
   }
 
   caf::error schema(vast::schema sch) {
+    if (sch.empty())
+      return make_error(ec::invalid_configuration, "no schema provided");
```

Comment:
Should this be checked at the call site already? In this function, we could make it a precondition.

Reply:
Sounds more error prone. `*x.begin()` without a check just shouldn't pass a review.

Reply:
Sorry if this was not clear, I'm talking about the *calling site* for the function `json::schema(..)`. That is where this check should go in my opinion. The reason: it's the same for all readers, so it can be factored.

I don't understand `*x.begin()` in the scope of this context.

Reply:
I understood that, and I don't think it generally makes sense to assume non-empty schemas for all input formats. Both source and reader command call `reader.schema` regardless of the format. Some input formats don't require a schema at all (e.g., NetFlow) and simply ignore this call, and they should continue working when using VAST without schemas.

> I don't understand `*x.begin()` in the scope of this context.

I was referring to the line immediately below the code you're commenting on, which is using `*sch.begin()`, which is a code smell to begin with.